app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,60 @@
+class PasswordResetsController < ApplicationController
+  before_action :load_user, only: %i(create edit update)
+  before_action :valid_user, :check_expiration, only: %i(edit update)
+
+  def new; end
+
+  def create
+    @user = User.find_by email: params[:password_reset][:email].downcase
+    if @user
+      @user.create_reset_digest
+      @user.send_password_reset_email
+      flash[:info] = t ".success_acct_sent"
+      redirect_to root_path
+    else
+      flash.now[:danger] = t ".fail_acct_sent"
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if params[:user][:password].empty?
+      @user.errors.add :password, "can't be empty"
+      render :edit
+    elsif @user.update user_params
+      log_in @user
+      @user.update_attribute :reset_digest, nil
+      flash[:success] = t ".success_psword_update"
+      redirect_to @user
+    else
+      flash.now[:danger] = t ".fail_ps_reset_mess"
+      render :edit
+    end
+  end
+
+  private
+
+  def user_params
+    params.require(:user).permit :password, :password_confirmation
+  end
+
+  def load_user
+    @user = User.find_by email: params[:email]
```

Comment:
add code check `user is nil` by
`return if @user`
`flash[:warning] = "user not found"`
`redirect_to root_path`